rust/src/lib.rs

```diff
@@ -355,36 +310,89 @@ pub unsafe extern "C" fn zcashlc_init_data_database(
     unwrap_exc_or(res, -1)
 }
 
-/// A struct that contains details about an account in the wallet.
+/// A struct that contains a 16-byte account uuid along with key derivation metadata for that
+/// account.
+///
+/// A returned value containing the all-zeros seed fingerprint and/or u32::MAX for the
+/// hd_account_index indicates that no derivation metadata is available.
 #[repr(C)]
 pub struct FfiAccount {
+    uuid_bytes: [u8; 16],
+    account_name: *mut c_char,
+    key_source: *mut c_char,
     seed_fingerprint: [u8; 32],
-    account_index: u32,
+    hd_account_index: u32,
+}
+
+impl FfiAccount {
+    const NOT_FOUND: FfiAccount = FfiAccount {
+        uuid_bytes: [0u8; 16],
```

Comment:
This works as a sentinel because this is not a valid UUID v4 encoding (possibly not a valid encoding for any UUID? I don't recall precisely, but at least for UUID v4 there are 6 bits that need to be set specifically), so it is distinguishable.